beacon_node/http_api/src/lib.rs

```diff
@@ -3671,17 +3648,22 @@ pub fn serve<T: BeaconChainTypes>(
                         // from what is sent back down the channel.
                         Ok(warp::reply::reply().into_response())
                     })
-                    .await?;
+                    .await;
+
+                if initial_result.is_err() {
+                    return task_spawner::convert_rejection(initial_result).await;
+                }
```

Comment:
This is where the silent drop of `Ok(error_response)` was happening previously.

Reply:
I'm missing something here, how do we get an `Ok(error_response)` from the above?

Reply:
Oh yeah, the previous spawn function would return that here:

https://github.com/sigp/lighthouse/blob/1fcada8a328be02987e8135bde0bd37f0eae59b0/beacon_node/http_api/src/task_spawner.rs#L77

Reply:
ah make sense, thanks!